pylzy/lzy/api/v1/utils/proxy_adapter.py

```diff
@@ -51,25 +50,27 @@ def materialize_if_sequence_of_lzy_proxies(obj: Any) -> Any:
     return result
 
 
-def lzy_proxy(entry_id: str, types: Sequence[Type], wflow: "LzyWorkflow", value: Either = Right(None)) -> Any:
+def lzy_proxy(
+        entry_id: str, types: Sequence[Type], wflow: "LzyWorkflow",
+        value: Either = Absence(ValueError("Missing default value"))
+) -> Any:
     async def __materialize() -> Any:
-
-        if isinstance(value, Left):
+        if isinstance(value, Result):
             return value.value
 
         data = await wflow.snapshot.get_data(entry_id)
-        if isinstance(data, Left):
+        if isinstance(data, Result):
             return data.value
 
         # noinspection PyProtectedMember
         await wflow._barrier()
 
         new_data = await wflow.snapshot.get_data(entry_id)
-        if isinstance(new_data, Left):
+        if isinstance(new_data, Result):
             return new_data.value
-        raise RuntimeError(
-            f"Cannot materialize data of entry with id={entry_id}, workflow={wflow.name}, cause={str(new_data.value)}"
-        )
+
+        assert isinstance(new_data, Absence)
+        raise new_data.cause
```

Comment:
Could you please check, will be traceback of original exception losed here?

Reply:
traceback from original exception will be saved